codegen/testserver/followschema/generated_test.go

```diff
@@ -78,4 +84,23 @@ func TestUnionFragments(t *testing.T) {
 		`, &resp)
 		require.NotEmpty(t, resp.ShapeUnion.Radius)
 	})
+
+	t.Run("without circle", func(t *testing.T) {
+		var resp struct {
+			Shapes []struct {
+				Length, Width float64
+			}
+		}
+		require.Empty(t, resp.Shapes)
+		c.MustPost(`query {
+			shapes {
+				... on Rectangle {
+					length
+					width
+				}
+			}
+		}
+		`, &resp)
+		require.Empty(t, resp.Shapes)
```

Comment:
Thats not compliant with the GraphQL spec. The resolver must return two empty objects (for the two circles) here. I also write the comment on the issue. Its not a but or inconvenience if the graphql server is returning empty objects in union or interface types. 

Thats part of the specification. https://spec.graphql.org/October2021/#sec-Unions 

and here is one quote: 

> interfaces must return the set of interfaces that an object implements (if none, interfaces must return the empty set).